README.md

```diff
@@ -1,6 +1,653 @@
+|---------------------------------------------------------------------------------------------------------------|
+| View Active Directory [on the plugin site](https://plugins.jenkins.io/active-directory) for more information. |
+
+Older versions of this plugin may not be safe to use. Please review the following warnings before using an older version:
+
+-   [Improper certificate validation](https://jenkins.io/security/advisory/2019-01-28/#SECURITY-859)
+-   [Man-in-the-middle vulnerability due to missing certificate check](https://jenkins.io/security/advisory/2017-03-20/)
+
+With this plugin, you can configure Jenkins to authenticate the username and the password through Active Directory.This plugin internally uses two very different implementations, depending on whether Jenkins is running on Windows or non-Windows and if you specify a domain.
+
+-   If Jenkins is running on a Windows machine and you do not specify a domain, that machine must be a member of the domain you wish to authenticate against. Jenkins uses ADSI to figure out all the details, so no additional configuration is required.
+-   If Jenkins is running on a non-Windows machine (or you specify one or more domains), then you need to tell Jenkins the name of Active Directory domain(s) to authenticate with. Jenkins then uses DNS SRV records and LDAP service of Active Directory to authenticate users.
+
+Jenkins recognizes all the groups in Active Directory that the user belongs to, so you can use those to make authorization decisions (for example you can choose the matrix-based security as the authorization strategy and perhaps allow "Domain Admins" to administer Jenkins).
+
+#### Active Directory Health Status
+
+Since the version 2.5 the AD plugin adds a ManagementLink to report a Health Status about the Domain and Domain controllers. In order to correctly use this feature, you should be logged-in into the instance and the cache should be disabled. Then, you will get:
+
+-   The Domain health
+    -   DNS resolution
+    -   Global Catalog
+    -   Ldap Catalog
+-   The Domain Controller Health
+    -   If the user can login into the DC
+    -   The Connection time
+    -   The total time in the lookup process
+
+![](docs/images/ad-managementLink-view.png)
+
+#### Fall-back user
+
+Since the version 2.5 of the AD plugin, you can define a user to fall back in case there is a communication issue between Jenkins and the AD server. On this way, this admin user can be used to continue administering Jenkins in case of communication issues, where usually you were following the link [Disable security](https://wiki.jenkins.io/display/JA/Disable+security). The password of this user is automatically synced with the Jenkins Internal Database by this feature. In order to configure this new feature you should enable \*Use Jenkins Internal Database\* in the AD configuration under Manage Jenkins → Configure Global Security and specify a SINGLE user by its username.
```

Comment:
```suggestion
Since the version 2.5 of the AD plugin, you can define a user to fall back in case there is a communication issue between Jenkins and the AD server.
On this way, this admin user can be used to continue administering Jenkins in case of communication issues, where usually you were following the link [Disable security](https://www.jenkins.io/doc/book/system-administration/security/#disabling-security).
The password of this user is automatically synced with the Jenkins Internal Database by this feature. In order to configure this new feature you should enable \*Use Jenkins Internal Database\* in the AD configuration under Manage Jenkins → Configure Global Security and specify a SINGLE user by its username.
```

Fixed the link + line breaks for easier reviews

Reply:
Fixed and pushed this change.